core/src/test/java/org/apache/iceberg/rest/TestRESTUtil.java

```diff
@@ -0,0 +1,106 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iceberg.rest;
+
+import java.util.List;
+import java.util.Map;
+import org.apache.iceberg.catalog.Namespace;
+import org.apache.iceberg.relocated.com.google.common.collect.ImmutableList;
+import org.apache.iceberg.relocated.com.google.common.collect.ImmutableMap;
+import org.assertj.core.api.Assertions;
+import org.junit.Test;
+
+public class TestRESTUtil {
+
+  static class LevelsAndURLEncodedString {
+    public String[] levels;
+    public String parameterEncodedForUrlUsage;
+
+    LevelsAndURLEncodedString(String[] levels, String parameterEncodedForUrlUsage) {
+      this.levels = levels;
+      this.parameterEncodedForUrlUsage = parameterEncodedForUrlUsage;
+    }
+  }
+
+  @Test
+  public void testFilterAndRemovePrefix() {
```

Comment:
Is there a performance concern vs using the `@Parameterized` annotation that instantiates a new test class?

The performance overhead of using `@Parameterized` is why I didn't do that here. I'll look into this. Thank you!

Reply:
no concern at all in terms of performance. I was rather thinking that the code could just look cleaner by separating input & expected result from the actual test code, since that makes reading the code easier. 

Note that in case you decide to use JUnit5 features here, you would have to add this to `build.gradle`:

```
project(':iceberg-core') {
  test {
    useJUnitPlatform()
  }
  ...
}
```

Reply:
Cool will check that out. Thank you.

Reply:
So for this specific test, I'm actually testing that the whole `input` map, when passed into the function, returns the `expected` map. Including that certain keys (which don't contain the prefix) are removed.

So I'm not sure if using `@ParameterizedTest` here really helps much. It will on `testTrailingSlash` as well as on the `testNamespaceUrlEncodeDecode` method below.

Reply:
Based on another comment, we'll evaluate junit 5 introduction to core possibly in another PR vs in a utilities class PR (for example, in case people don't cherrypick this PR in a fork they maintain etc).